hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -418,9 +434,11 @@ private void initThreadPools(Configuration conf) {
     unboundedThreadPool = new ThreadPoolExecutor(
         maxThreads, Integer.MAX_VALUE,
```

Comment:
This is where I had suggested we drop the first argument to 0 (for boundedThreadPool too) as that's core threads, not max threads. Otherwise we actually lock ourselves at the maximum and grow from there. Only if you rebuild and retest anyway - otherwise I'll submit a patch once this is in to avoid further conflicts...